### PR TITLE
Feature/post remove ingredients

### PR DESCRIPTION
**Pull request by Adelle-Pitsas**
What does this PR do?
[x] adds `removeIngredientsFromPantry` to event listener if the button's text is `Cook Recipe!` 
[x] `removeIngredientsFromPantry` creates the post requests and invokes `postAll` with the requests as arguments.
[x] adds `removePantryIngredients` to the user class to handle the removal of recipes that have been cooked
[x] updates user's pantry - data model
[x] updates user's pantry display - DOM

Where should your reviewer start?
New Post is on `line 123` of `scripts.js`

What (if any) features are you implementing?
- Post to API to remove ingredients when user cooks recipe ([trello ticket](https://trello.com/c/zUEQdtay/84-add-remove-ingredients-from-a-pantry-in-scriptsjs))

What (if anything) did you refactor?

Were there any issues that arose?

Is there anything that you need from your teammate?

Any other comments, questions, or concerns to add?

### CODE REVIEW

**Comment by tricia-holmes** [comment]
Question :

When we remove ingredients from the pantry, should we completely remove from the pantry the ingredients that have `0 amount`? Right now after we post the remove, our pantry has 0 ingredients that display in the pantry. I feel like because we don’t list all the ingredients that a pantry could have in the user’s pantry that we should splice out 0s. I wanted to see though makes sense as a user to everyone else?

**Comment by tricia-holmes** [comment]
I tried this in this `PR` and `main` and it looks like this already exists in main. When `missing ingredients button` is clicked multiple times, the `add all ingredients button` gets duplicated each time. 


https://user-images.githubusercontent.com/108706408/200151253-2e0f6d28-8b48-40c2-894f-24bcbdf0f80b.mp4

**Comment by tricia-holmes** [comment]
This can also be a seperate ticket , but we need to `add an event listener` to the `cook recipe button` that appears after a user successfully adds pantry ingredients.

**Comment by Adelle-Pitsas** [comment]
> 

Yes! I saw that as I was working yesterday afternoon, and forgot to make a ticket for it! Thanks for making the trello card!

**Comment by Adelle-Pitsas** [comment]
I'll work on removing the zeroed out ingredients in our pantry display and re-push! Thanks!

**Comment by tricia-holmes** [comment]
Everything looks great! Merging this in now!

File: src/classes/Ingredient.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ class Ingredient {
     allIngredientData.name = fromIngredients.name;
     allIngredientData.cost = fromIngredients.estimatedCostInCents;
     allIngredientData.amount =
-      ingredientSource.amount || ingredientSource.quantity.amount;
+      ingredientSource.amount || ingredientSource.quantity?.amount;
     if (ingredientSource.quantity) {
       allIngredientData.unit = ingredientSource.quantity.unit;
```
Comment (tricia-holmes) [inline src/classes/Ingredient.js:23]: The `optional chaining` works great here! We no longer are getting a null for the amount when we load the user.

**Pull request merged**